c_src/trans_nif.c

```diff
@@ -89,7 +89,7 @@ derivate(ErlNifEnv* env, int argc, const ERL_NIF_TERM argv[])
   if (count < 1) // can't be empty
     return enif_make_badarg(env);
 
-  if (! (target = (decimal*) enif_make_new_binary(env, bin.size - sizeof(decimal), &r)))
+  if (! (target = (decimal*) enif_make_new_binary(env, (count - 1) * sizeof(decimal), &r)))
```

Comment:
I think this makes sense, the derivate will always be 1 element shorter, `bin.size` was probably a bug here.